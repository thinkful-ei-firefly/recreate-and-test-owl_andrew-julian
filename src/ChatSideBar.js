import React from 'react';
import ChatMessages from './ChatMessages';
import cuid from 'cuid';

function ChatSideBar(props) {
    return (
      <div className="ChatSideBar">
          {props.chatEvents.map((chat) =>
              <ChatMessages
                    key={cuid()}
                    participantId={chat.participantId}
                    type={chat.type}
                    message={chat.message}
                    time={chat.time}
                    timestamp={chat.timestamp}

                  />
              )}
      </div>
    );
  }

export default ChatSideBar;