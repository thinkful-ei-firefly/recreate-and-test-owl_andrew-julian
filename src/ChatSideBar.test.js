import React from 'react';
import ReactDOM from 'react-dom';
import ChatSideBar from './ChatSideBar';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <ChatSideBar 
    chatEvents={[{
        participantId: 3,
        type: 'message',
        message: 'Hello world',
        time: 1548852646559,
        timestamp: 1548852484247
    },
    {
        participantId: 2,
        type: 'thumbs-up',
        timestamp: 1548852484247
    }]}
        />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<ChatSideBar 
        chatEvents={[{
            participantId: 3,
            type: 'message',
            message: 'Hello world',
            time: 1548852646559,
            timestamp: 1548852484247
        },
        {
            participantId: 2,
            type: 'thumbs-up',
            timestamp: 1548852484247
        }]}
            />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });

