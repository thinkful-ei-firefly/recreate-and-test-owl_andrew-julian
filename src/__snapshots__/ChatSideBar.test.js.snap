// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="ChatSideBar"
>
  <div>
    &lt;img class="chatAvatar" src=url&gt;&lt;p class="chatPrefix"&gt;srting 1/3 19&lt;/p&gt;&lt;p class="chatMessage"&gt;Hello world&lt;/p&gt;
  </div>
  <div>
    &lt;p className="notification"&gt;srting gave a thumbs up&lt;/p&gt;&lt;span id='thumbsup' className="tui-icon icon-thumbsup"&gt;&lt;/span&gt;
  </div>
</div>
`;
